pkg/models/metrics/metrics.go

```diff
@@ -19,1259 +19,782 @@
 package metrics
 
 import (
+	"fmt"
+	"github.com/json-iterator/go"
 	"k8s.io/klog"
-	"kubesphere.io/kubesphere/pkg/informers"
+	"kubesphere.io/kubesphere/pkg/models/workspaces"
+	cs "kubesphere.io/kubesphere/pkg/simple/client"
 	"net/url"
 	"regexp"
-	"runtime/debug"
-	"sort"
 	"strings"
 	"sync"
 	"time"
 
-	"github.com/json-iterator/go"
-
-	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/labels"
-
-	"kubesphere.io/kubesphere/pkg/models/workspaces"
-	cs "kubesphere.io/kubesphere/pkg/simple/client"
+	"kubesphere.io/kubesphere/pkg/simple/client/prometheus"
```

Comment:
here, we import `kubesphere.io/kubesphere/pkg/simple/client` to create p8s client, but we also have to import `kubesphere.io/kubesphere/pkg/simple/client/prometheus` to use type defined in it. This would be easily cause cyclic import. We encourage to put type definitions in `pkg/api` to create a neutral package to avoid this.